src/modules/app-db/repositories/nft.repository.ts

```diff
@@ -98,6 +100,32 @@ export class NftRepository {
         walletAddress: walletAddress
       }
     });
+  }   
+
+  //Removes NFT
+  async removeNFT(userId: string , nftId: string) {
+

```

Comment:
instead of doing this you can the following:
Add onDelete: 'SET NULL' to @OneToOne nft relationship in artwork.entity.
Add the same for trialMint in user.entity.
To apply changes to db you will need to create a migration:
npm run migration:generate ./src/migrations/remove-nft-constraint
After that you can just call:
this.nfts.remove(this.nfts.create({ id: nftId })) and related entities will be automatically updated

Reply:
Perfect idea!